### PR TITLE
fix: fixed bug where the feature store does not deserialize KV values…

### PR DESCRIPTION
**Pull request by yusinto**
* removed { type: 'json' } from KV calls to return default string values from the KV
* expose deserializePoll and use it to deserialize string KV values
* fixed unit tests

### CODE REVIEW

File: packages/sdk/cloudflare/package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "start": "rimraf dist && yarn tsw",
     "lint": "eslint . --ext .ts",
     "prettier": "prettier --write '**/*.@(js|ts|tsx|json|css)' --ignore-path ../../../.prettierignore",
-    "test": "NODE_OPTIONS=\"--experimental-vm-modules --no-warnings\" jest --ci --runInBand --coverage",
```
Comment (yusinto) [inline packages/sdk/cloudflare/package.json:36]: Coverage is nice but noisy. I'm removing it for now.

File: packages/sdk/cloudflare/src/createFeatureStore.test.ts [hunk 2]
```diff
@@ -7,6 +7,16 @@ import * as testData from './utils/testData.json';
 describe('createFeatureStore', () => {
   const sdkKey = 'sdkKey';
   const kvKey = `LD-Env-${sdkKey}`;
+  const {
+    testFlag1: { debugEventsUntilDate: d1, ...testFlag1Subset },
+    testFlag2: { debugEventsUntilDate: d2, ...testFlag2Subset },
+    testFlag3: { debugEventsUntilDate: d3, ...testFlag3Subset },
+  } = testData.flags;
```
Comment (yusinto) [inline packages/sdk/cloudflare/src/createFeatureStore.test.ts:14]: The deserializePoll function adds `debugEventsUntilDate` which doesn't exist in `testData.json`. So I'm omitting them here to be ignored in tests.

File: packages/sdk/cloudflare/src/createFeatureStore.test.ts [hunk 3]
```diff
@@ -18,7 +28,7 @@ describe('createFeatureStore', () => {
   let asyncFeatureStore: AsyncStoreFacade;
 
   beforeEach(() => {
-    mockGet.mockImplementation(() => Promise.resolve(testData));
+    mockGet.mockImplementation(() => Promise.resolve(JSON.stringify(testData)));
```
Comment (yusinto) [inline packages/sdk/cloudflare/src/createFeatureStore.test.ts:31]: KV calls now return default strings.

File: packages/sdk/cloudflare/example/package.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
   "module": "./dist/index.mjs",
   "packageManager": "yarn@3.4.1",
   "dependencies": {
-    "@launchdarkly/cloudflare-server-sdk": "0.0.1"
+    "@launchdarkly/cloudflare-server-sdk": "2.0.1"
```
Comment (kinyoklion) [inline packages/sdk/cloudflare/example/package.json:8]: Not a problem for right now, but I wonder if we can get release please to update this file as well.

File: packages/sdk/cloudflare/package.json (hunk 1 above)
Comment (kinyoklion) [reply packages/sdk/cloudflare/package.json:36]: I usually just make a second command "coverage". Which runs basically "yarn test -- --coverage" (or however you forward the parameter)

File: packages/sdk/cloudflare/src/createFeatureStore.test.ts (hunk 2 above)
Comment (kinyoklion) [reply packages/sdk/cloudflare/src/createFeatureStore.test.ts:14]: I see it in the test data?
image

I don't see something in `deserializePoll` that would affect them.

File: packages/shared/sdk-server/src/store/serialization.ts [hunk 5]
```diff
@@ -22,6 +22,7 @@ export function reviver(this: any, key: string, value: any): any {
 }
 
 interface FlagsAndSegments {
+  [key: string]: { [name: string]: Flag } | { [name: string]: Segment };
```
Comment (kinyoklion) [inline packages/shared/sdk-server/src/store/serialization.ts:25]: I know you probably did this to appease typescript, but this really changes the shape of this object. Which makes it less clear what is correct code in other places. I think I have some thoughts about how to handle this at the calling side.

File: packages/sdk/cloudflare/example/package.json (hunk 4 above)
Comment (yusinto) [reply packages/sdk/cloudflare/example/package.json:8]: That will be the best. Any idea how?

File: packages/sdk/cloudflare/package.json (hunk 1 above)
Comment (yusinto) [reply packages/sdk/cloudflare/package.json:36]: done

File: packages/sdk/cloudflare/src/createFeatureStore.test.ts (hunk 2 above)
Comment (yusinto) [reply packages/sdk/cloudflare/src/createFeatureStore.test.ts:14]: I'll investigate.

File: packages/sdk/cloudflare/example/package.json (hunk 4 above)
Comment (kinyoklion) [reply packages/sdk/cloudflare/example/package.json:8]: I do not know. I am looking into making some other build robustness changes. Maybe I will figure it out. Nothing important to solve right this second.

File: packages/sdk/cloudflare/src/createFeatureStore.test.ts (hunk 2 above)
Comment (yusinto) [reply packages/sdk/cloudflare/src/createFeatureStore.test.ts]: I set debugEventsUntilDate to 2000 so it's non null so it's not stripped by serializer.ts.

File: packages/shared/sdk-server/src/store/serialization.ts (hunk 5 above)
Comment (yusinto) [reply packages/shared/sdk-server/src/store/serialization.ts]: Merged #111 which contained your suggestions. Thank you.

**Review by kinyoklion — APPROVED** [review]

**Pull request merged**